### PR TITLE
Accessibility: Manage Household contact search navigation

### PR DESCRIPTION
**Pull request by dcaballeroSf**
# Critical Changes

# Changes
In Manage Household page, on contact lookup field accessibility capabilities were added:
   You can navigate on the result list in the lookup using Arrows keys and select an element with enter key
# Issues Closed

# Community Ideas Delivered

# New Metadata
- Aura Event:
    - HH_KeypressEvent

- Aura Component:
    - HH_NewFocussedElement
    - HH_ReachFooter

# Deleted Metadata

### CODE REVIEW

**Comment by dcaballeroSf** [comment]
[WI](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07B0000008T2sMIAS/view)

File: src/aura/autocomplete/autocompleteHelper.js [hunk 1]
```diff
@@ -71,5 +76,106 @@
...
+        
+        if (currentFocussedElement == null || currentFocussedElement == '') {
+            if (elements[elements.length - 1]) {
+                currentFocussedElement = elements[elements.length - 1].value.Id
+                listbox.setAttribute('aria-activedescendant', currentFocussedElement);
+                helper.fireNewItemFocus(currentFocussedElement);
+                return;
+            }
+        }
+        var newFocussedElement = null;
```
Comment (npsp-reedestockton) [inline src/aura/autocomplete/autocompleteHelper.js:145]: This is not used until the for loop below.  Can we move this to immediately before that for loop and add a blank line here?

File: src/aura/autocomplete/autocompleteHelper.js [hunk 2]
```diff
@@ -71,5 +76,106 @@
...
+                if (showFooter) {
+                    helper.fireFocusOnFooter(component, listbox);
+                    return;
+                }
+            }
+            
+        }
+
+        var newFocussedElement = null; 
+
```
Comment (npsp-reedestockton) [inline src/aura/autocomplete/autocompleteHelper.js:105]: Can we remove this blank line?  newFocussedElement is an integral part of the for loop below, so I don't think the blank line helps with understanding what is happening here.

File: src/aura/autocomplete/autocompleteHelper.js [hunk 3]
```diff
@@ -71,5 +76,106 @@
...
+                        listbox.setAttribute('aria-activedescendant', newFocussedElement);
+                        break;
+                    }
+                }
+            }
+        }
+        
+        helper.fireNewItemFocus(newFocussedElement);
+    },
+    // Handles the pressing of the arrow up key.
```
Comment (npsp-reedestockton) [inline src/aura/autocomplete/autocompleteHelper.js:130]: Can we add a blank line before these two comment lines?

File: src/package.xml [hunk 4]
```diff
@@ -812,6 +812,9 @@
         <members>HH_Container</members>
         <members>HH_ContainerLoadedEvent</members>
         <members>HH_HouseholdSavedEvent</members>
+        <members>HH_keypressEvent</members>
```
Comment (npsp-reedestockton) [inline src/package.xml:815]: Our convention for naming module specific aura bundles is MOD_CapitalizedName, so HH_KeypressEvent, etc. would be more in line with our practice.

**Comment by dcaballeroSf** [comment]
Adding Sam for reviewing new metadata description

File: src/aura/autocomplete/autocompleteHelper.js [hunk 5]
```diff
@@ -71,5 +76,109 @@
...
+                }
+            }
+        }
+        
+        helper.fireNewItemFocus(newFocussedElement);
+    },
+
+    // Handles the pressing of the arrow up key.
+    // It updates the aria-activedescendant and notify the child elements
+
```
Comment (npsp-reedestockton) [inline src/aura/autocomplete/autocompleteHelper.js:133]: Can we remove this blank line?

File: src/aura/HH_AutoCompleteOption/HH_AutoCompleteOptionController.js [hunk 6]
```diff
@@ -2,4 +2,40 @@
...
     },
+    // Handle keypress event on the elements of the list
+    handleKeyPress : function(component, event, helper) {
+        /*
+          Need to check if the key pressed is enter. If so, the element has to trigger the selected element event
+          if no, an event has to be raised to the autocomplete component to manage the new focus
+          */
+
+        //stopping propagation to avoid lastpass bug
+        event.preventDefault()
```
Comment (RadSFDeveloper) [inline src/aura/HH_AutoCompleteOption/HH_AutoCompleteOptionController.js:13]: Missing a `;` after statement on this and the next line.

File: src/aura/HH_AutoCompleteOption/HH_AutoCompleteOptionController.js [hunk 7]
```diff
@@ -2,4 +2,40 @@
...
+        event.stopPropagation()
+        var key = event.key;
+        if (key == 'Enter') {
+            helper.fireOptionSelectedEvent(component);;
+            return;
+        }
+        //
```
Comment (RadSFDeveloper) [inline src/aura/HH_AutoCompleteOption/HH_AutoCompleteOptionController.js:20]: Let's remove `\\` to represent a new line in the code.

File: src/aura/HH_AutoCompleteOption/HH_AutoCompleteOptionController.js [hunk 8]
```diff
@@ -2,4 +2,40 @@
...
+        var event = $A.get("e.c:HH_KeypressEvent");
+        event.setParams({
+            "keyPressed" : key
+        });
+        event.fire();
+     },
+    
+     // Event that handles the new focus on the right element after the autocomple component has calculated the next element
```
Comment (RadSFDeveloper) [inline src/aura/HH_AutoCompleteOption/HH_AutoCompleteOptionController.js:28]: Typo in "autocomple".
Let's use `/* ` for *all new* methods descriptions.

File: src/aura/HH_AutoCompleteOption/HH_AutoCompleteOptionController.js [hunk 9]
```diff
@@ -2,4 +2,40 @@
...
+     handleNewFocussedElement : function(component, event, helper) {
+        var id = event.getParam('id');
+        if (id == null) return;
```
Comment (RadSFDeveloper) [inline src/aura/HH_AutoCompleteOption/HH_AutoCompleteOptionController.js:31]: `if` code blocks should be enclosed in `{` and `}`.

File: src/aura/HH_KeypressEvent/HH_KeypressEvent.evt-meta.xml [hunk 10]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<AuraDefinitionBundle xmlns="http://soap.sforce.com/2006/04/metadata">
+    <apiVersion>49.0</apiVersion>
+    <description>A Lightning Event Bundle</description>
```
Comment (RadSFDeveloper) [inline src/aura/HH_KeypressEvent/HH_KeypressEvent.evt-meta.xml:4]: Let's add description why and when event is used.

File: src/aura/HH_NewFocussedElement/HH_NewFocussedElement.evt-meta.xml [hunk 11]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<AuraDefinitionBundle xmlns="http://soap.sforce.com/2006/04/metadata">
+    <apiVersion>49.0</apiVersion>
+    <description>A Lightning Event Bundle</description>
```
Comment (RadSFDeveloper) [inline src/aura/HH_NewFocussedElement/HH_NewFocussedElement.evt-meta.xml:4]: Let's add description for new LCs.

**Comment by RadSFDeveloper** [comment]
Hi @rjanjanam  -- can you please review the failed robot tests and check if it is intermittent issue?

File: src/aura/HH_AutoCompleteOption/HH_AutoCompleteOptionController.js (hunk 6 above)
Comment (RadSFDeveloper) [reply src/aura/HH_AutoCompleteOption/HH_AutoCompleteOptionController.js]: This comment is still outstanding, @diego-caballero .

File: src/aura/HH_NewFocussedElement/HH_NewFocussedElement.evt-meta.xml (hunk 11 above)
Comment (RadSFDeveloper) [reply src/aura/HH_NewFocussedElement/HH_NewFocussedElement.evt-meta.xml]: This comment is still oustanding

**Comment by skristem** [comment]
@RadSFDeveloper the failed robot test is due to a chrome issue. In such cases, the solution is to rebuild. I just restarted the build.

File: src/aura/HH_NewFocussedElement/HH_NewFocussedElement.evt-meta.xml (hunk 11 above)
Comment (dcaballeroSf) [reply src/aura/HH_NewFocussedElement/HH_NewFocussedElement.evt-meta.xml]: done.

**Review by RadSFDeveloper — APPROVED** [review]
Looks good, thanks @diego-caballero !

**Comment by RadSFDeveloper** [comment]
@CrystalPreston-Watson , the Manage Household accessibility WI is ready for testing.

**Comment by andrewyu-salesforce** [comment]
@npsp-reedestockton Can you review and approve the change. @CrystalPreston-Watson  has finished testing it and everything looks good on her side

**Review by ravensisco — APPROVED** [review]
Testing has passed by A11y QE

**Review by npsp-reedestockton — APPROVED** [review]
Looks good.

**Pull request merged**